### PR TITLE
Sen-990 -- style guide compliance dynamodb-marshaler

### PR DESCRIPTION
**Pull request by nackjicholson**
Made the code and tests comply with https://github.com/CascadeEnergy/node-style-guide

**To Test**

Verify the examples still work.

`node examples/example-marshal.js`
`node examples/example-unmarshal.js`

### CODE REVIEW

**Comment by nackjicholson** [comment]
Build failed, but looks like an internal travis thing when it tried to install node. WIll look into it.

File: .jshintrc [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
+  "freeze": true,
+  "indent": 2,
+  "newcap": true,
+  "quotmark": "single",
+  "maxdepth": 3,
+  "maxstatements": 15,
+  "maxlen": 80,
+  "eqnull": true,
+  "node": true,
+  "es5": true,
```
Comment (cberube) [inline .jshintrc]: Did you want to remove the `es5` here?

File: lib/commands/unmarshalMap.js [hunk 2]
```diff
@@ -3,12 +3,12 @@
...
+  }
 
-    return _.reduce(item.M, function(result, value, key) {
-        result[key] = unmarshal(value);
-        return result;
-    }, {});
+  return _.reduce(item.M, function(result, value, key) {
+    result[key] = unmarshal(value);
+    return result;
+  }, {});
```
Comment (cberube) [inline lib/commands/unmarshalMap.js]: Would `mapValues` work here, e.g.: `return _.mapValues(item.M, unmarshal);`?

File: .jshintrc (hunk 1 above)
Comment (nackjicholson) [inline .jshintrc]: Yup, no longer required in jshint.

File: test/commands/unmarshalPassthrough.js [hunk 3]
```diff
@@ -1,23 +1,27 @@
...
-    }, function(item, expected) {
-        it('should return the value for the pass through key', function() {
-            unmarshalPassThrough(item).should.eql(expected);
-        });
+  withData({
+    'with an "S" item': [{S: 'foo'}, 'foo'],
+    'with "SS" item': [{SS: ['foo', 'bar']}, ['foo', 'bar']],
+    'with "B" item': [{B: 'binary'}, 'binary'],
+    'with "BS" item': [{BS: ['binary1', 'binary2']}, ['binary1', 'binary2']],
+    'with "BOOL" item': [{BOOL: true}, true]
```
Comment (nwarlen) [inline test/commands/unmarshalPassthrough.js]: withData adds `with` to the name of the test automatically so these test results are outputting `with with an "s" item`, not a big deal but makes reading the test results a little odd.

From travis:
```
unmarshalPassThrough
  ✓ should return undefined if item does not contain a key which is a pass through type 
    with with an "S" item
  ✓ should return the value for the pass through key 
    with with "SS" item
  ✓ should return the value for the pass through key 
    with with "B" item
  ✓ should return the value for the pass through key 
    with with "BS" item
  ✓ should return the value for the pass through key 
    with with "BOOL" item
  ✓ should return the value for the pass through key 
```

Comment (nackjicholson) [inline test/commands/unmarshalPassthrough.js]: Damn, I thought I had that looking more english like at some point.

File: .jshintrc (hunk 1 above)
Comment (cberube) [inline .jshintrc]: I think this is the last outstanding comment on this PR.

Comment (nackjicholson) [inline .jshintrc]: It was removed. see fc80806

**Comment by cberube** [comment]
:ok:

**Pull request merged**